### DISCUSSION TITLE
API Github list contributors

### DISCUSSION THREAD

**Discussion by joan-saum** [community/community#24355] [Apps, API and Webhooks]
<p>Hi everyone,</p>
<p>I’m currently working with the GitHub API and i think there is a problem (or the mistake is mine but i don’t know where)</p>
<p>I tried to know the number of contributors for a specific repository thanks to this endpoint :</p>
<p><a href="https://developer.github.com/v3/repos/#list-contributors" rel="nofollow noopener">https://developer.github.com/v3/repos/#list-contributors</a></p>
<p>Work well on small repository but seems to be wrong on bigger repository</p>
<p>For example, let’s take the repository twbs/bootstrap</p>
<p>On the <a href="https://github.com/twbs/bootstrap" rel="nofollow noopener">github page</a>, it display 1007 contributors</p>
<p>If i try this one : <a href="https://api.github.com/repos/twbs/bootstrap/contributors?per_page=1&amp;page=900" rel="nofollow noopener">https://api.github.com/repos/twbs/bootstrap/contributors?per_page=1&amp;page=900</a></p>
<p>I want the 900th contributors. But nothing appeared. Actually, it work below the page 381, but then, up to 381, nothing appear. Like if there is only 381 contributors</p>
<p>So at first i tought the problem was with anonymous contributors. So i tried to add them like this :</p>
<p><a href="https://api.github.com/repos/twbs/bootstrap/contributors?per_page=1&amp;page=900&amp;anon=true" rel="nofollow noopener">https://api.github.com/repos/twbs/bootstrap/contributors?per_page=1&amp;page=900&amp;anon=true</a></p>
<p>(I added the query anon and set it to true)</p>
<p>And it worked. And actually it worked too much because, if i try to have the 1100th contributors like this:</p>
<p><a href="https://api.github.com/repos/twbs/bootstrap/contributors?per_page=1&amp;page=1100&amp;anon=true" rel="nofollow noopener">https://api.github.com/repos/twbs/bootstrap/contributors?per_page=1&amp;page=1100&amp;anon=true</a></p>
<p>API return me a response with someone but the github page display 1007 contributors.</p>
<p>Am i doing something wrong ?</p>
<p>Thanks for help!</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>No, you’re not doing anything wrong. As it states in the documentation:</p>
<blockquote>
<p>To improve performance, only the first 500 author email addresses in the repository link to GitHub users. The rest will appear as anonymous contributors without associated GitHub user information.</p>
</blockquote>
<p>Therefore, on larger repos, you may not be able to replicate the exact figure that we show on the website.</p>

**Comment by joan-saum** [comment]
<p>Thank you for your answer lee-dohm</p>
<p>I read this sentence in the documentation but i don’t care about GitHub user information so i tought it will work anyway with anonymous contributors</p>
<p>But i don’t understand, if i count anonymous contributors, why do i have more contributors than show on the GitHub page ? Who has the correct number ? The API or the GitHub page ?</p>

**Comment by lee-dohm** [comment]
<p>The GitHub page has the correct number.</p>
<p>Here’s why the API might give a higher number than you expect:</p>
<ol>
<li>If the repository has more than 500 contributors (for example 501, 500 associated with GitHub information and 1 by definition anonymous)</li>
<li>One or more of the least active contributors made commits with more than one email address (for example 2 different email addresses)</li>
</ol>
<p>Using the example numbers, the GitHub page would read as 501 contributors but you would see 502 records.</p>

**Comment by joan-saum** [comment]
<p>Got it!</p>
<p>Thank you for your explanations !</p>

**Comment by miriyusifli** [comment]
<p>Hi, I have the same problem. The GitHub page shows 834 contributors.<br>
I am trying to get this number using API. API shows that repo has 1002 contributors.</p>
<p><a href="https://api.github.com/repos/apple/swift/contributors?page=1002&amp;per_page=1&amp;anon=true" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/repos/apple/swift/contributors?page=1002&amp;per_page=1&amp;anon=true</a></p>
<p>I have listed all contributors and removed duplicates by “name”. After that, the list still has 910 contributors. Why do I still have more than 834 contributors?</p>
<p>Could you please give me an idea of how can I get the exact number on the GitHub page?</p>
<p>I would be thankful if you explain who is Anonymous user. I didn’t understand why some users are anonymous. As far as I know, the first 500 contributors are linked with the corresponding user, and others are anonymous. Why anonymous users exist before the 500th contributor? For example, the 350th contributor is anonymous<br>
<a href="https://api.github.com/repos/apple/swift/contributors?page=350&amp;per_page=1&amp;anon=true" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/repos/apple/swift/contributors?page=350&amp;per_page=1&amp;anon=true</a><br>
Thanks in advance.</p>

**Comment by lee-dohm** [comment]
<p>A user can be “anonymous” if there is no GitHub user associated with a given email address. And the reason that your number still may not match the one given by the UI is because the same GitHub user may have contributed using multiple email addresses. This is why I said <a href="https://github.community/t/api-github-list-contributors/13520/2">above</a>:</p>
<blockquote>
<p>on larger repos, you may not be able to replicate the exact figure that we show on the website.</p>
</blockquote>
<p>The API simply doesn’t return the information you need in order to be able to replicate the number we show.</p>

**Comment by miriyusifli** [comment]
<p>Thanks for the explaination :smile:</p>
